python/lsst/daf/butler/core/logging.py

```diff
@@ -21,11 +21,185 @@
 
 from __future__ import annotations
 
-__all__ = ("VERBOSE",)
+__all__ = ("VERBOSE", "ButlerLogRecords", "ButlerLogRecordHandler")
 
 import logging
+import datetime
+from typing import List, Union, Optional, ClassVar, Iterable, Iterator
+
+from logging import LogRecord, StreamHandler
+from pydantic import BaseModel, ValidationError
 
 VERBOSE = (logging.INFO + logging.DEBUG) // 2
 """Verbose log level"""
 
+_LONG_LOG_FORMAT = "%(levelname)s %(asctime)s %(name)s %(filename)s:%(lineno)s - %(message)s"
+"""Default format for log records."""
+
 logging.addLevelName(VERBOSE, "VERBOSE")
+
+
+class ButlerLogRecord(BaseModel):
+    """A model representing a `logging.LogRecord`.
+
+    A `~logging.LogRecord` always uses the current time in its record
+    when recreated and that makes it impossible to use it as a
+    serialization format. Instead have a local representation of a
```

Comment:
You can overwrite LogRecord attributes after creating it, you just need to know which ones, and there is a `logging.makeLogRecord()` method which looks like it was specifically designed for supporting LogRecord serialization (probably as a dict). Pickling LogRecod directly is probably not the right thing to do, in particular new MDC attribute needs some care, you probably want to serialize it a as a simple dict, to avoid dependency on `log.lsst.MDCDict` class.

Reply:
Right, that's all why I created `ButlerLogRecord` so that we can have the flexibility to deal with MDC.